website/src/components/SchemaFormatTabs.astro

```diff
@@ -131,6 +151,241 @@ function extractCode(props: CodeBlockConfig): string {
+            continue;
+          }
+          
+          const exampleValue = generateJsonFromSchemaObject(propSchema as JsonSchema, visitedRefs, rootSchema);
+          
+          // Include required fields and optional fields that have valid examples
+          if (isRequired || (exampleValue !== null && exampleValue !== undefined)) {
+            obj[key] = exampleValue;
+          }
+        }
+      }
+      return obj;
+    }
+
+    case 'array':
+      if (schema.items) {
+        const itemExample = generateJsonFromSchemaObject(schema.items as JsonSchema, visitedRefs, rootSchema);
+        return [itemExample];
+      }
+      return [];
+
+    case 'string':
+      if (schema.enum) {
+        return schema.enum[0];
+      }
+      if (schema.format === 'date-time') {
+        return new Date().toISOString().split('T')[0];
+      }
+      if (schema.format === 'uri') {
+        return "https://example.com";
+      }
+      if (schema.examples && schema.examples.length > 0) {
+        return schema.examples[0];
+      }
+      return "Sample string value";
+
+    case 'integer':
+      if (schema.examples && schema.examples.length > 0) {
+        return schema.examples[0];
+      }
+      return 10;
+
+    case 'number':
+      if (schema.examples && schema.examples.length > 0) {
+        return schema.examples[0];
+      }
+      return 100.50;
+
+    case 'boolean':
+      return true;
+
+    default:
+      return null;
+  }
+}
+
+/**
+ * Resolves a schema reference by loading the referenced schema file or resolving internal definitions.
+ * @param ref - The reference string (e.g., "OppStatus.yaml", "#/$defs/RecordEvent")
+ * @param rootSchema - The root schema object for resolving internal $defs
+ * @returns The parsed schema object, or null if the reference cannot be resolved
+ */
+function resolveSchemaReference(ref: string, rootSchema?: JsonSchema): JsonSchema | null {
+  // Handle internal $defs references (e.g., "#/$defs/RecordEvent")
+  if (ref.startsWith('#/$defs/')) {
+    if (rootSchema && rootSchema.$defs) {
+      const defName = ref.replace('#/$defs/', '');
+      return rootSchema.$defs[defName] || null;
+    }
+    return null;
+  }
+  try {
+    const websiteRoot = process.cwd();
+    const repoRoot = dirname(websiteRoot);
+    
+    // Handle different reference formats
+    let filePath: string;
+    if (ref.startsWith('website/public/schemas/yaml/')) {
+      // Full path reference
+      filePath = join(repoRoot, ref);
+    } else if (ref.includes('/')) {
+      // Relative path reference
+      filePath = join(repoRoot, 'website/public/schemas/yaml', ref);
+    } else {
+      // Just filename reference
+      filePath = join(repoRoot, 'website/public/schemas/yaml', ref);
+    }
+    
+    const content = readFileSync(filePath, "utf-8");
+    return yaml.load(content, { schema: yaml.CORE_SCHEMA }) as JsonSchema;
+  } catch {
+    // If we can't load the referenced schema, return null
+    return null;
+  }
```

Comment:
We're already collecting and resolving all of these schemas in [validation.ts](https://github.com/HHS/simpler-grants-protocol/blob/c2dab45b9bbad9576b6160bba2802bad18151537/website/src/lib/validation.ts#L13-L15)

If you create an `ajv` instance using `createAjvWithSchemas()` (or export that instance) then you can replace this with:

```ts
  // Use AJV to resolve external references
  try {
    const resolvedSchema = ajv.getSchema(ref);
    return (resolvedSchema?.schema as JsonSchema) || null;
  } catch {
    return null;
  }
```

The other bonus of using the existing ajv instance is that we can actually validate the generated code against the JSON schema to make sure it's valid